Taglierino.hs

```diff
@@ -345,6 +345,26 @@ receive = do
       return $ LTS.Action l body
     return $ LTS.Body actions
 
+insertFreshL :: [LTS.LabelPart] -> LTS.Label
+insertFreshL args = LTS.Label $ [LTS.Simple "insertfresh"] ++ args
+
+insertFresh :: Term -> Proc Bool
+insertFresh r = do
+  ProcRState{..} <- ask
+  case S.lookupIndex r psAllowed of
+    Just mi -> 
+      Proc $ ContT $ \k -> do
+        s <- get
```

Comment:
Why are you reading the state and writing it back?

Reply:
I tried to keep it same as receive since insertFresh is kind of like receive where it branches to different state.